internal/bufferpool/bufferpool.go

```diff
@@ -18,32 +18,30 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-// Package buffers provides strongly-typed functions to interact with a shared
-// pool of byte slices.
-//
-// It's used heavily inside zap, but callers may also take advantage of it;
-// it's particularly useful when implementing json.Marshaler, text.Marshaler,
-// and similar interfaces.
-package buffers
+// Package bufferpool provides strongly-typed functions to interact with a shared
+// pool of byte buffers.
+package bufferpool
```

Comment:
The `Buffer` type needs to be non-internal, since it's part of the exported interfaces. The pool, however, should remain internal.